pandas/core/groupby.py

```diff
@@ -2790,6 +2790,10 @@ def true_and_notnull(x, *args, **kwargs):
     def nunique(self, dropna=True):
         """ Returns number of unique elements in the group """
         ids, _, _ = self.grouper.group_info
+        
+        if len(ids) == 0:  # bufix for 12553
+            return Series([])
```

Comment:
This should just be `self._constructor`, so objects that inherit can keep their type